sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -984,6 +984,30 @@ abstract class AnsiCastSuiteBase extends CastSuiteBase {
     }
   }
 
+  test("ANSI mode: cast string to date with parse error") {
+    val activeConf = conf
+    new ParVector(ALL_TIMEZONES.toVector).foreach { zid =>
```

Comment:
Do we really need to check all time zones? Would `DateTimeTestUtils.outstandingZoneIds` be enough?

Reply:
+1

Reply:
This is from the test case for timestamp. 
+1, let me update both.